### PR TITLE
Feature/SOF-7056

### PR DESCRIPTION
**Pull request by VsevolodX**

### CODE REVIEW

File: src/other/pyodide-loader/PyodideLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
```
Comment (timurbazhirov) [inline src/other/pyodide-loader/PyodideLoader.tsx]: pyodide as the folder name

File: src/other/pyodide-loader/PyodideLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            script.onerror = () => reject(new Error(`Failed to load script ${src}`));
+            document.body.appendChild(script);
+        });
+    };
+
+    const initializePyodide = async () => {
+        try {
+            await createScriptTag(url);
+            window.pyodide = await window.loadPyodide();
+            const loadedPyodide = window.pyodide;
```
Comment (timurbazhirov) [inline src/other/pyodide-loader/PyodideLoader.tsx]: Should be assigned to a prop variable

Comment (VsevolodX) [reply src/other/pyodide-loader/PyodideLoader.tsx]: Props in React are readonly. The correct approach is to pass a callback function from parent, that will get this pyodide instance after it's fully loaded. We use this pattern everywhere.

File: src/other/pyodide/PyodideLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Loads Pyodide in window.pyodide and initializes it with micropip to allow for packages installation.
+ * @param url The URL to load Pyodide from.
+ * @param getPyodide A callback that receives the Pyodide instance once it is loaded.
+ * @param triggerLoad A variable that triggers the loading of Pyodide. If true, Pyodide will be loaded.
+ * @constructor
+ */
+function PyodideLoader({
+    url = defaultSourceUrl,
+    getPyodide,
```
Comment (timurbazhirov) [inline src/other/pyodide/PyodideLoader.tsx]: let's call it `onLoad`

File: src/other/pyodide/PyodideLoader.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+declare global {
+    interface Window {
+        // @ts-ignore
+        loadPyodide: () => Promise<any>;
+        // @ts-ignore
+        pyodide: any;
+    }
+}
+
+const defaultSourceUrl = "https://cdn.jsdelivr.net/pyodide/v0.24.0/full/pyodide.js";
```
Comment (k0stik) [inline src/other/pyodide/PyodideLoader.tsx:20]: I checked the size of this file. It's really small - only 7KB. Why don't we just install it with npm?
Plus it has implementation in TS: https://github.com/pyodide/pyodide/blob/main/src/js/api.ts so we probably would be able to remove `any` everywhere in this file

Comment (VsevolodX) [reply src/other/pyodide/PyodideLoader.tsx:20]: As we saw live, NPM version meant to work on Node.js, here we want to start Pyodide fully in the browser, thus the use of CDN which provides exactly that. But I'll import Pyodide from NPM for types annotations to remove all `any`s.

Comment (timurbazhirov) [reply src/other/pyodide/PyodideLoader.tsx:20]: As discussed, using CDN

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**